base/uk.ac.stfc.isis.ibex.ui.configserver/src/uk/ac/stfc/isis/ibex/ui/configserver/commands/helpers/EditConfigHelper.java

```diff
@@ -65,12 +65,25 @@ protected void openDialog(EditableConfiguration config, boolean isCurrent,
             if (dialog.doAsComponent()) {
                 server.saveAsComponent().uncheckedWrite(dialog.getComponent());
             } else {
-                if (isCurrent && dialog.switchConfigOnSaveAs()) {
-                    server.setCurrentConfig().uncheckedWrite(dialog.getConfig());
+                if (isCurrent) {
+                    if (dialog.calledSwitchConfigOnSaveAs()) {
+                        if (dialog.switchConfigOnSaveAs()) {
+                            server.setCurrentConfig().uncheckedWrite(dialog.getConfig());
```

Comment:
The nesting here is getting rather deep (5 levels of if-statements). It would be time to break this down into some separate functions.

Actually I think this whole if-statement structure needs to be refactored into separate methods - it is not easy to follow at the moment. Probably the best thing to do is to define to separate methods in the model - `setCurrentConfig` and `saveAndSwitchConfig` (or something along those lines), these can then be unit tested.